WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -359,34 +359,13 @@ private void buildAndShowNotificationFromNoteData(Context context, Bundle data)
             if (!buildNoteObjectFromPNPayloadAndSaveIt(data)) {
                 // PN payload doesn't have the note or there was an error.
                 // Retrieve the Note obj by calling the REST API
-
-                WordPress.getRestClientUtilsV1_1().getNotification(
-                        wpcomNoteID,
+                NotificationsActions.downloadNoteAndUpdateDB(wpcomNoteID,
```

Comment:
Actually I wasn't able to reproduce the issue above without manually forcing a double update for the received note.
Anyway, there was a bug in the code that writes notes into the database, where the on conflict resolution strategy seems to delete/insert a new row in the table, instead of updating it. In this case, the delete triggers are not called and the UI is not updated accordingly.
Weird that we didn't notice it before, since this PR didn't change the DB code.
It should be fixed now.